src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,138 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private Entry<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Entry[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
+        int hash = hash(key.hashCode());
+        int index = indexFor(hash, table.length);
+
+        for (Entry<K, V> entry = table[index]; entry != null; entry = entry.next) {
+            if ((entry.hash == hash) && (key.equals(entry.key))) {
+                entry.value = value;
+                return;
+            }
```

Comment:
Instead of hash comparison in such blocks I'd recommend to use 
entry.key = key
in order not to make separate method and calls for null and notNull key scenario